src/lib/comms/sol-lwm2m.c

```diff
@@ -4733,11 +4733,6 @@ handle_unknown_bootstrap_resource(void *data, struct sol_coap_server *server,
     sol_coap_header_set_code(resp, header_code);
     r = sol_coap_send_packet(server, resp, cliaddr);
 
-    if (header_code == SOL_COAP_RESPONSE_CODE_DELETED) {
-        obj_instance_clear(client, obj_ctx, obj_instance);
```

Comment:
Two things here:

1. I believe I got confused a bit with the commits, as we already talked before. I will be more careful next time. Anyway, if you look at the changes from "all commits", you'll see I've actually removed this from here.
2. I believe, as per your comment down below regarding not removing the instance from the list, I need to put it somewhere else. I'll try to figure it out.